### PR TITLE
[Feature] Add PoseWarper dataset

### PR DESCRIPTION
**Pull request by liqikai9**
## Motivation

Add [PoseWarper](https://arxiv.org/abs/1906.04016) dataset part.

## Modification

Add `$mmpose/mmpose/datasets/datasets/top_down/topdown_posetrack18_multiframe_dataset.py`

Modify `$mmpose/mmpose/datasets/datasets/__init__.py` and `$mmpose/mmpose/datasets/datasets/top_down/__init__.py` to add the new class.

Modify `$mmpose/mmpose/datasets/pipelines/loading.py` to support loading multiple images for an instance, modify the `LoadImageFromFile` class.

Modify `$mmpose/mmpose/datasets/pipelines/top_down_transform.py` to support transforming multiple images, including `TopDownRandomFlip`, `TopDownAffine`.

### CODE REVIEW

**Comment by jin-s13** [comment]
Please git rebase properly

**Comment by jin-s13** [comment]
Currently, there are some unrelated changes.
e.g. configs/_base_/datasets/300w.py

File: mmpose/datasets/datasets/top_down/topdown_posetrack18_multiframe_dataset.py [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+
+            image_files = []
+            cur_image_file = os.path.join(self.img_prefix,
+                                          self.id2name[img_id])
+            image_files.append(cur_image_file)
+
+            prev_nm = file_name.split('/')[-1]
+            ref_idx = int(prev_nm.replace('.jpg', ''))
+
+            # select the frame indices
```
[the remark this answers is not in the document]
Comment (luminxu) [reply mmpose/datasets/datasets/top_down/topdown_posetrack18_multiframe_dataset.py:184]: Good idea! In this way, it seems that we can use the original posetrack dataset for both image-based and video-based training/testing. Then this dataset is not needed anymore?

File: mmpose/datasets/datasets/top_down/topdown_posetrack18_multiframe_dataset.py [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+
+            deltas = self.timestep_delta
+            for delta in deltas:
+                sup_idx = ref_idx + delta
+                sup_idx = np.clip(sup_idx, 0, nframes - 1)
+                sup_image_file = cur_image_file.replace(
+                    prev_nm,
+                    str(sup_idx).zfill(6) + '.jpg')
+
+                if os.path.exists(sup_image_file):
```
[the remark this answers is not in the document]
Comment (luminxu) [reply mmpose/datasets/datasets/top_down/topdown_posetrack18_multiframe_dataset.py:283]: Thanks for your explanation and I understand the reason now. A warning would be fine. If the file does not exist, we use the current frame as the official code and print out a warning.

File: mmpose/datasets/datasets/top_down/topdown_posetrack18_multiframe_dataset.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liqikai9) [reply mmpose/datasets/datasets/top_down/topdown_posetrack18_multiframe_dataset.py:184]: The original `TopDownPoseTrack18Dataset` lacked some meta-information, e.g., `nframes` and `frame_id` that are closely related to video-based algorithms. Further, considering that the datasets and pipelines will be refactored in the OpenMMLab 2.0 design, I will use this dataset for now, and maybe we can modify them when the refactor scheme is determined.

**Review by luminxu — APPROVED** [review]
Please rebase properly before merging.

**Comment by jin-s13** [comment]
Shall we close this PR?

**Comment by liqikai9** [comment]
> Shall we close this PR?
I think we can close this PR and move the related code to #1006

**Pull request closed without merging**